src/solidity/stateDecoder.js

```diff
@@ -0,0 +1,233 @@
+'use strict'
+var BN = require('ethereumjs-util').BN
+var utileth = require('ethereumjs-util')
+var varUtil = require('./variable')
+
+module.exports = {
+  decodeArray: function (type, storage, location) {
+    return this.decodeArrayItems(type, storage, 0, location)
+  },
+
+  decodeArrayItems: function (type, storage, depth, location) {
```

Comment:
I think it would be nice to improve ethereumjs-abi and use it here. I'd like to change the types into ES6 classes with encoding/decoding, as well as having composite types (such as array).

@yann300 are you interested in this?